docs/how_to_guide/handling_composite_bases.md

```diff
@@ -0,0 +1,156 @@
+
+```{code-cell} ipython3
+import nemos as nmo
+
+# define a composite basis
+add = nmo.basis.RaisedCosineLinearEval(5, label="input1") + nmo.basis.BSplineEval(6, label="input2")
+
+# `add` stores the two 1dimensional bases as attributes
+print(add)
+print(add.basis1)
+print(add.basis2)
+```
+
+Composing even more, will result in more nesting of attributes.
+
+```{code-cell} ipython3
+add = add + nmo.basis.MSplineEval(4, label="input3")
+print(add)
+print(add.basis1.basis1)
+print(add.basis1.basis2)
+print(add.basis2)
+```
+
+## Retrieving Basis Components and Their Parameters
+In principle, nesting makes the process of retrieving or setting the parameters of individual components quite cumbersome.  
+
+```{code-cell} ipython3
+# retreive the number of basis funciton for input2 basis
+add.basis1.basis2.n_basis_funcs
+```
+
+However, if you associated a label to the basis, you can use it to get the corresponding basis element.
+
+```{code-cell} ipython3
+add["input2"]
+```
+
+And its parameters can be easily accessed.
+
+```{code-cell} ipython3
+add["input2"].n_basis_funcs
+```
+
+This works for any sub-element, including the one that are composite.
+
+```{code-cell} ipython3
+# get input1 + input2
+add["(input1 + input2)"]
+```
+
+Note that the label of this composite basis is assigned automatically. You can overwrite that with a custom label.
+
+```{code-cell} ipython3
+add["(input1 + input2)"].label = "my_custom_label"
+add
+```
+
+A label can be specified at initialization if the composite basis is defined directly.
+
+```{code-cell} ipython3
+nmo.basis.AdditiveBasis(
+    nmo.basis.BSplineEval(5),
+    nmo.basis.MSplineEval(5), 
+    label="my_custom_label"
+)
+```
+
+And if you are asking yourself what happens when two bases with the same label are composed, well, this results in an error.
+This guarantees that the labels are always unique and you can always retrieve a basis using its label.
+
+```{code-cell} ipython3
+:tags: [raises-exception]
+
+nmo.basis.BSplineEval(5, label="x") + nmo.basis.MSplineEval(5, label="x")
+```
+
+## Get and Set Composite Basis Parameters
+
+When working with composite bases, often times one wants to re-configurate specific components. Again, the easiest way to achieve this is labeling each element and using the label to retrieve the basis.
+
+```{code-cell} ipython3
+# get the basis function parameter
+print(add["input2"].n_basis_funcs)
+
+# set a new value for the parameter
+add["input2"].n_basis_funcs = 8
+print(add["input2"].n_basis_funcs)
+```
+
+This change is reflected on the composite basis.
+
+```{code-cell} ipython3
+# check that the input2 basis has now 8 basis funcs
+add
+```
+
+### Cross-validating Composite Basis Parameters
```

Comment:
Needs a different title: this section is actually about interacting with them in a sklearn-type way, using `set_params()`, so title should reflect that.

Reply:
and then there should be a comment saying something like "this is how sklearn interacts with basis objects, and so enables cross-validation, etc"